test/index.test.js

```diff
@@ -187,18 +220,12 @@ describe("iw-mysql-wrapper", function () {
 						transactionScope.query("INSERT INTO `iw_mysql_wrapper_test` VALUES (1);"),
 						transactionScope.query("BAD SQL;")
 					]);
-				}).then(function () {
-
-					var err = new Error("Transaction should have failed!");
-					err.code = "E_DID_NOT_FAIL";
-					throw err;
-
-				}).catch(function verifyAfterRollback(err) {
+				}).then(neverCallMe("Query should throw a parse error.")).catch(function verifyAfterRollback(err) {
 
 					expect(err.code).to.eql("ER_PARSE_ERROR");
 					return Q.all([
-						savedScope.query("SELECT * FROM `iw_mysql_wrapper_test`").then(function (rows) {
-							expect(rows).to.eql([], "Transaction scope: should be rolled back");
+						savedScope.query("SELECT 1;").then(neverCallMe("Transaction should be closed.")).catch(function (err) {
+							expect(err.code).to.eql("E_TRANSACTION_CLOSED");
```

Comment:
As for "test name should define what's expected" - what's expected is written out explicitly in the test code, we can argue to death about naming here, but I don't think that would give us any value, other than the fun of the discussion. Do you honestly think that the test name is problematic or is it pure dogma?